presto-base-jdbc/src/main/java/io/prestosql/plugin/jdbc/BaseJdbcClient.java

```diff
@@ -389,6 +389,24 @@ protected ResultSet getColumns(JdbcTableHandle tableHandle, DatabaseMetaData met
                 DISABLE_PUSHDOWN));
     }
 
+    // Type mismatch should be eventually caught by JdbcRecordCursor. Early verification helps fail fast.
+    protected void verifyExpressionType(ConnectorSession session, JdbcExpression jdbcExpression, Type expectedType)
+    {
+        try (Connection connection = connectionFactory.openConnection(session)) {
```

Comment:
Opening a connection just to call verify, sounds like a big price just for a defensive check. Can we avoid opening a connection?

Reply:
That's a valid concern and i had it too.
The problem is, `toPrestoType` takes a `Connection`. I added it to avoid taking opening additional connections when mapping array types in PostgreSQL. Of course, this is a niche use case.
The middle ground would be to pass `Optional<Connection>` or `Supplier<Connection>`, or yet have a lazy connection (a bit too clever maybe?)
cc @losipiuk @electrum 

Also, in case we do not do any of these _yet_, what do you think about this PR?

Reply:
With connection pooling it should not be that expensive.
Though I think it would be nice to change it to `Supplier<Connection>`. Can you file issue for that?

Reply:
`Supplier<Connection>` will be hard to use when you actuially need it. (who closes this?)
Lazy connection wouldn't have this problem.